dot/rpc/modules/system.go

```diff
@@ -135,3 +157,74 @@ func (sm *SystemModule) NodeRoles(r *http.Request, req *EmptyRequest, res *[]int
 	*res = resultArray
 	return nil
 }
+
+// AccountNextIndex Returns the next valid index (aka. nonce) for given account.
+func (sm *SystemModule) AccountNextIndex(r *http.Request, req *StringRequest, res *U64Response) error {
+	if req == nil || len(req.String) == 0 {
+		return errors.New("Account address must be valid")
+	}
+	addressPubKey := crypto.PublicAddressToByteArray(common.Address(req.String))
+
+	// check pending transactions for extrinsics singed by addressPubKey
+	pending := sm.txStateAPI.Pending()
+	nonce := uint64(0)
+	found := false
+	for _, v := range pending {
+		var ext ctypes.Extrinsic
+		err := ctypes.DecodeFromBytes(v.Extrinsic[1:], &ext)
+		if err != nil {
+			return err
+		}
+		extSigner, err := common.HexToBytes(fmt.Sprintf("0x%x", ext.Signature.Signer.AsAccountID))
+		if err != nil {
+			return err
+		}
+		if bytes.Equal(extSigner, addressPubKey) {
+			found = true
+			sigNonce := big.Int(ext.Signature.Nonce)
+			if sigNonce.Uint64() > nonce {
+				nonce = sigNonce.Uint64()
```

Comment:
wouldn't the nonce always be greater unless it's 0, in which case it's fine to directly set it?

Reply:
Yes, but there may be more than one tx signed by that user in the tx pool, so did this to make sure we have the maximum nonce.